src/lib.rs

```diff
@@ -39,36 +82,73 @@ fn gen_checksum(v: &[impl AsRef<[u8]>]) -> String {
   out.join("")
 }
 
-#[derive(Debug)]
-pub struct LockfileError(String);
-
-impl std::fmt::Display for LockfileError {
-  fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
-    f.write_str(&self.0)
-  }
+#[derive(Debug, Error)]
+pub enum LockfileError {
+  #[error(transparent)]
+  IntegrityCheckFailed(#[from] IntegrityCheckFailedError),
 }
 
-impl std::error::Error for LockfileError {}
+#[derive(Debug, Error)]
+#[error("Integrity check failed for npm package: \"{package_display_id}\". Unable to verify that the package
+is the same as when the lockfile was generated.
+
+Actual: {actual}
+Expected: {expected}
+
+This could be caused by:
+  * the lock file may be corrupt
+  * the source itself may be corrupt
+
+Use \"--lock-write\" flag to regenerate the lockfile at \"{filename}\".",
+)]
+pub struct IntegrityCheckFailedError {
```

Comment:
Question: will we surface these errors one-by-one or aggregate them and display them all at once?

Reply:
One-by-one as it currently does. It might be good to improve this in the future.